### PR TITLE
Dynamically expose issuer and subject of Compass cert on the info endpoint

### PR DESCRIPTION
**Pull request by nyordanoff**
**Description**

Currently, our info endpoint exposes the subject and the issuer of the Compass certificate but this information is statically configured in the environment of the director and returned when the endpoint is hit.

This PR configures the info endpoint to load the issuer and subject dynamically from the certificate itself (reading the cert from the cert cache every time the endpoint is hit) rather than using hardcoded env values.

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/info/handler.go [hunk 1]
```diff
@@ -24,24 +33,46 @@ type responseData struct {
...
+	parsedClientCert, err := x509.ParseCertificate(clientCert.Certificate[0])
+	if err != nil {
+		return responseData{}, errors.New("error while parsing client certificate")
+	}
+
+	certIssuer := replaceDelimiter(parsedClientCert.Issuer.String())
+	certSubject := replaceDelimiter(parsedClientCert.Subject.String())
+
+	return responseData{
+		Issuer:     certIssuer,
```
[the remark this answers is not in the document]
Comment (nyordanoff) [reply components/director/internal/info/handler.go:69]: From my POV it is a bit clearer how it is now. It is not some 'central' flow so I guess we are fine if we leave it as is.

File: components/director/internal/info/handler.go [hunk 2]
```diff
@@ -24,24 +33,46 @@ type responseData struct {
...
-func prepareResponseData(c Config) responseData {
-	return responseData{
-		Issuer:     c.Issuer,
-		Subject:    c.Subject,
-		RootCA:     c.RootCA,
-		OrdVersion: ord.SpecVersion,
-	}
-}
-
 // NewInfoHandler returns handler which gives information about the CMP client certificate
```
Comment (la4ezar) [inline components/director/internal/info/handler.go:36]: Adapt godoc.

**Review by la4ezar — APPROVED** [review]

**Pull request merged**